lms/djangoapps/grades/tests/test_models.py

```diff
@@ -238,3 +240,46 @@ def test_update_or_create_grade(self, already_created):
         if already_created:
             self.assertEquals(created_grade.id, updated_grade.id)
             self.assertEquals(created_grade.earned_all, 6)
+
+
+@ddt.ddt
+class PersistentCourseGradesTest(GradesModelTestCase):
+    """
+    Tests the PersistentCourseGrade model.
+    """
+    def setUp(self):
+        super(PersistentCourseGradesTest, self).setUp()
+        self.block_records = BlockRecordList([self.record_a, self.record_b], self.course_key)
+        self.params = {
+            "user_id": 12345,
+            "course_id": self.course_key,
+            "course_version": "JoeMcEwing",
+            "subtree_edited_timestamp": "2016-08-01 18:53:24.354741",
+            "percent_grade": "77.7",
+            "letter_grade": "Great job",
+        }
+
+    @ddt.data(True, False)
+    def test_update_or_create(self, already_created):
```

Comment:
Coding exercise: Refactor this test code to 2 smaller tests - with each test testing a specific scenario (already created versus not already created).  It may simplify the tests and allow you to remove all the `if` conditionals.  If not DRY, a helper function may be needed.